packages/components/dialog/src/DialogContent.tsx

```diff
@@ -1,33 +1,44 @@
 import * as RadixDialog from '@radix-ui/react-dialog'
-import { forwardRef, type ReactElement, type Ref } from 'react'
+import { forwardRef, type ReactElement, type Ref, useLayoutEffect } from 'react'
 
 import {
   dialogContentStyles,
   type DialogContentStylesProps,
   dialogContentWrapperStyles,
 } from './DialogContent.styles'
+import { useDialog } from './DialogContext'
 
 export type ContentProps = RadixDialog.DialogContentProps & DialogContentStylesProps
 
 export const Content = forwardRef(
   (
     { children, className, size = 'md', ...rest }: ContentProps,
     ref: Ref<HTMLDivElement>
-  ): ReactElement => (
-    <div className={dialogContentWrapperStyles()}>
-      <RadixDialog.Content
-        data-spark-component="dialog-content"
-        ref={ref}
-        className={dialogContentStyles({
-          size,
-          className,
-        })}
-        {...rest}
-      >
-        {children}
-      </RadixDialog.Content>
-    </div>
-  )
+  ): ReactElement => {
+    const { setIsFullScreen } = useDialog()
+
+    useLayoutEffect(() => {
```

Comment:
```suggestion
    useEffect(() => {
```

Reply:
Why ? I prefer `useLayoutEffect` it's good for this case: https://react.dev/reference/react/useLayoutEffect

Reply:
> Usage: Measuring layout before the browser repaints the screen
That is not the case here, you don't read values from an HTML element 🤷

Reply:
No by I need this value updated in the state before the component is rendered, not after.

Reply:
Yes, we can discuss it this morning 👍